src/vcpkg/base/unicode.cpp

```diff
@@ -222,28 +222,34 @@ namespace vcpkg::Unicode
             Checks::unreachable(VCPKG_LINE_INFO);
         }
 
-        if (next_ == last_)
+        const auto old_next = next_;
+        const auto last = last_;
+        if (old_next == last)
         {
             current_ = end_of_file;
+            pointer_to_current_ = last;
             return utf8_errc::NoError;
         }
 
         char32_t code_point;
-        auto new_next = utf8_decode_code_point(next_, last_, code_point);
-        if (new_next.second != utf8_errc::NoError)
+        auto err = utf8_decode_code_point(next_, last, code_point);
+        if (err != utf8_errc::NoError)
         {
-            *this = sentinel();
-            return new_next.second;
+            current_ = end_of_file;
+            pointer_to_current_ = last;
+            return err;
         }
 
         if (utf16_is_trailing_surrogate_code_point(code_point) && utf16_is_leading_surrogate_code_point(current_))
         {
-            *this = sentinel();
+            current_ = end_of_file;
+            pointer_to_current_ = last;
+            next_ = last;
             return utf8_errc::PairedSurrogates;
         }
 
-        next_ = new_next.first;
         current_ = code_point;
+        pointer_to_current_ = old_next;
         return utf8_errc::NoError;
     }
 
```

Comment:
Can we get rid of `operator++`? Or is it in use somewhere else?

Reply:
I'm pretty sure lots of places use this as an iterator still

Reply:
Yeah this is used all over.